### ISSUE TITLE
Crashing After Highlight

### ISSUE THREAD

**Issue by harikeerthipati** [FolioReader/FolioReaderKit#186]
Hi, After Highlited the text, again MenuController is showing hightlight options only instead of showing colors options. Then I tapped on highlight option, app is getting crashed here. I m getting this issue in latest code which is committed 2 days ago.

**Comment by hebertialmeida** [comment]
Hey, please check the latest master version.

**Comment by harikeerthipati** [comment]
Crash issue fixed. But the old issue coming again. We have two types of text selection options for uiwebview. One is double tap and another is long press on text. Double tap works fine. But when selected text using long press, first it is showing initial MenuController(consists Copy, Play, Highlight etc..) options fine. But when selected next time it is directly showing colors options and if we press the color also nothing highlighting there. Please fix it.

**Comment by hebertialmeida** [comment]
Could you make a video of that, simulating the problem? Will help me on that.

**Comment by harikeerthipati** [comment]
Hi,

https://drive.google.com/file/d/0B-ALJ1rRn1aUaWZ1Q2trdENzdGM/view?usp=sharing <https://drive.google.com/file/d/0B-ALJ1rRn1aUaWZ1Q2trdENzdGM/view?usp=sharing>

Please find the video in above google drive link. In this you can find where exactly issue generating.

1) At first i have double taped on text, this case everything works fine. I mean first I have selected Highlight, after that color options will appear. when selected color, that color is applying to text background.
2) Next I have long pressed on the text. Here I have pressed Highlight option. But I did’t see any color options here.
3) When I tried to select text for third time, I m unable to see Highlight option. Here app showing directly color options. When we select on a color it is highlighting previously highlighted text.


Thanks & Regards
Harikrishna.K

Mobifusion Pvt. Ltd.
(o): +1.510.870.1546 | (f): +1.510.217.5013
www.mobifusion.com <http://www.mobifusion.com/>



Silicon Valley | Miami | Boston | Kuala Lumpur | Delhi | Hyderabad
The information contained in this communication is intended for the use of the designated recipients named above. If the reader of this communication is not the intended recipient, you are hereby notified that you have received this communication in error, and that any review, dissemination, distribution or copying of this communication is strictly prohibited. If you have received this communication in error, please notify Mobifusion immediately by telephone at +1.510.870.1546 and delete this email. Thank you.

**Comment by revolter** [comment]
I can't reproduce the original issue/crash :(

**Comment by revolter** [comment]
@harikeerthipati, Any possibility for you to revert to the old version and record a video where you reproduce the crash? It will help me solve both issues.

**Comment by PravinNagargoje** [comment]
@harikeerthipati  I have fixed your issue. New changes i have made as follows : 
 1) tap and hold to see main menu(highlight, play etc..) and the nit will ask for color selection. after there was a issue that color menu not disappearing, I have fixed that also.
2) one tap on highlighted text it will directly show you color menu and two taps or tap and hold on highlighted text it will take you to the main menu(highlight , play etc..). 
Hope it will solve your issue.

I have sent pull request for this issue.
